frontend/src/components/Duration/Duration.tsx

```diff
@@ -2,14 +2,23 @@ import React, { useState } from 'react'
 import { useInterval } from '../../hooks/useInterval'
 import humanize from 'humanize-duration'
 
-export const dateOptions = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' }
+export const dateOptions: Intl.DateTimeFormatOptions = {
+  weekday: 'long',
+  year: 'numeric',
+  month: 'long',
+  day: 'numeric',
+}
 
-export const Duration: React.FC<{ startTime?: number; ago?: boolean }> = ({ startTime, ago = false }) => {
-  const [now, setNow] = useState(Date.now())
+export const Duration: React.FC<{ startTime?: number; endTime?: number; ago?: boolean }> = ({
+  startTime,
+  endTime = Date.now(),
+  ago = false,
+}) => {
+  const [now, setNow] = useState<number>(endTime)
   const aDay = 1000 * 60 * 60 * 24
 
   useInterval(() => {
```

Comment:
I think that it should be a set timeout not setInterval, maybe double-check the hooks because we want to setup. 

I am not 100% sure

Reply:
Yea interval is correct for this because it's repeating forever will be a little lighter on resources. And the useInterval will clean up when the component is teared down